public/app/plugins/datasource/cloudwatch/components/MetricStatEditor/MetricStatEditor.tsx

```diff
@@ -44,15 +62,15 @@ export function MetricStatEditor({
               options={namespaces}
               onChange={({ value: namespace }) => {
                 if (namespace) {
-                  onQueryChange({ ...query, namespace });
+                  onNamespaceChange({ ...query, namespace });
                 }
               }}
             />
           </EditorField>
           <EditorField label="Metric name" width={16}>
             <Select
               aria-label="Metric name"
-              value={query.metricName}
+              value={query.metricName || null}
```

Comment:
This is needed as the `select` component needs a `null` value to reset (does not do so for `''`)